packages/react/src/template.tsx

```diff
@@ -99,6 +109,9 @@ export function isActionsDataState(obj: DataState): obj is ActionsDataState {
 export function isActionDataState(obj: any): obj is ActionDataState {
   return obj !== null && typeof obj === 'object' && '$' in obj && obj.$ === 'action'
 }
+function isStateDataState(obj: any): obj is StateDataState {
+  return obj !== null && typeof obj === 'object' && '$' in obj && obj.$ === 'state'
```

Comment:
btw I don't think it is necessary to check for `'$' in obj`

Reply:
TypeScript complains otherwise for `obj.$` :(